inference-engine/thirdparty/clDNN/runtime/ocl/ocl_ext.hpp

```diff
@@ -240,529 +157,558 @@ typedef CL_API_ENTRY cl_mem(CL_API_CALL * PFN_clCreateFromMediaSurfaceINTEL)(
+                // to correctly capture each underlying CL object
+                for (size_type i = 0; i < ids.size(); i++) {
+                    (*devices)[i] = Device(ids[i], true);
                 }
+            }
 
-                // set up acquire/release extensions
-                SharedSurfLock::Init(object_);
-                ImageVA::Init(object_);
+            // set up acquire/release extensions
+            SharedSurfLock::Init(object_);
+            ImageVA::Init(object_);
 #ifdef WIN32
-                BufferDX::Init(object_);
+            BufferDX::Init(object_);
 #endif
-            }
-            return CL_SUCCESS;
-        }
-    };
-
-    /*
-        UsmPointer requires associated context to free it.
-        Simple wrapper class for usm allocated pointer.
-    */
-    class UsmHolder {
-    public:
-        explicit UsmHolder(Context& ctx, void* ptr) : _ctx(ctx), _ptr(ptr), deleter(nullptr) {
-            deleter = load_entrypoint<clMemFreeINTEL_fn>(_ctx.get(), "clMemFreeINTEL");
-            if (!deleter) {
-                throw std::runtime_error("clMemFreeINTEL is nullptr in UsmHolder");
-            }
         }
-        void* ptr() { return _ptr; }
-        ~UsmHolder() {
-            deleter(_ctx.get(), _ptr);
+        return CL_SUCCESS;
+    }
+};
+
+class UsmHelper {
+public:
+    explicit UsmHelper(const cl::Context& ctx, const cl::Device device, bool use_usm) : _ctx(ctx), _device(device) {
+        if (use_usm) {
+            _host_mem_alloc_fn             = try_load_entrypoint<clHostMemAllocINTEL_fn>(_ctx.get(), "clHostMemAllocINTEL");
+            _shared_mem_alloc_fn           = try_load_entrypoint<clSharedMemAllocINTEL_fn>(_ctx.get(), "clSharedMemAllocINTEL");
+            _device_mem_alloc_fn           = try_load_entrypoint<clDeviceMemAllocINTEL_fn>(_ctx.get(), "clDeviceMemAllocINTEL");
+            _mem_free_fn                   = try_load_entrypoint<clMemFreeINTEL_fn>(_ctx.get(), "clMemFreeINTEL");
+            _set_kernel_arg_mem_pointer_fn = try_load_entrypoint<clSetKernelArgMemPointerINTEL_fn>(_ctx.get(), "clSetKernelArgMemPointerINTEL");
+            _enqueue_memcpy_fn             = try_load_entrypoint<clEnqueueMemcpyINTEL_fn>(_ctx.get(), "clEnqueueMemcpyINTEL");
+            _enqueue_mem_fill_fn           = try_load_entrypoint<clEnqueueMemFillINTEL_fn>(_ctx.get(), "clEnqueueMemFillINTEL");
+            _enqueue_memset_fn             = try_load_entrypoint<clEnqueueMemsetINTEL_fn>(_ctx.get(), "clEnqueueMemsetINTEL");
+        } else {
+            _host_mem_alloc_fn             = nullptr;
+            _shared_mem_alloc_fn           = nullptr;
+            _device_mem_alloc_fn           = nullptr;
+            _mem_free_fn                   = nullptr;
+            _set_kernel_arg_mem_pointer_fn = nullptr;
+            _enqueue_memcpy_fn             = nullptr;
+            _enqueue_mem_fill_fn           = nullptr;
+            _enqueue_memset_fn             = nullptr;
         }
```

Comment:
My minor comment is to move initialization of these variables by null to in-class member initialization and remove from constructor

Reply:
Good point. fixed